legacy/bootloader/usb.c

```
@@ -198,445 +396,1148 @@ static int should_keep_storage(int old_was_signed,
+  if (!validate_protocol_header(buf)) {
+    return secfalse;
+  }
+  *msg_id = (buf[3] << 8) | buf[4];
+  *msg_size = ((uint32_t)buf[5] << 24) | ((uint32_t)buf[6] << 16) |
+              ((uint32_t)buf[7] << 8) | buf[8];
+  return sectrue;
+}
 
-  p_buf = packet_buf;
+// Helper function: Handle error and update state
+static void handle_error(usbd_device *dev, uint8_t error_code,
```

Clang (14.0.6)

[warning] parameter 'dev' is unused

(misc-unused-parameters)